src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -401,30 +391,31 @@ class SftpServer
 
     void handle_close(sftp_client_message msg)
     {
-        auto handle = handle_from_msg(msg);
-        if (!handle)
+        const auto id = sftp_handle(msg->sftp, msg->handle);
+
+        auto erased = open_file_handles.erase(id);
+        erased += open_dir_handles.erase(id);
```

Comment:
Is there any way to tell what type (file or dir) is being closed instead of just calling erase on both unordered_maps?  Or would two erase() calls be just a quick as figuring out the type and then calling erase() on the particular unordered_map?

Reply:
Yeah you would still have to lookup the id in one map then the other to figure out which type it was. So same amount of lookups.